cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/preview/DefaultPreviewRunner.java

```diff
@@ -139,6 +149,25 @@ public void startPreview(PreviewRequest<?> previewRequest) throws Exception {
       programId, previewConfig == null ? Collections.<String, String>emptyMap() : previewConfig.getRuntimeArgs(),
       false);
 
+    // Only have timer if there is a timeout setting.
+    if (previewConfig != null && previewConfig.getTimeout() != null) {
```

Comment:
Yes. If user directly pass it using REST call instead of starting a preview through UI.